src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3530,25 +3525,13 @@ private unsafe void ParseRemaining()
             // Note that we don't support one-letter schemes that will be put into a DOS path bucket
 
             int end = idx;
-            while (end < length && uriString[end] != ':')
+            while (end < uriString.Length && uriString[end] != ':')
             {
                 ++end;
             }
 
-            // NB: On 64-bits we will use less optimized code from CheckSchemeSyntax()
-            //
-            if (IntPtr.Size == 4)
-            {
-                // long = 4chars: The minimal size of a known scheme is 2 + ':'
-                if (end != length && end >= idx + 2 &&
-                    CheckKnownSchemes((long*)(uriString + idx), end - idx, ref syntax))
-                {
-                    return end + 1;
-                }
-            }
-
             //NB: A string must have at least 3 characters and at least 1 before ':'
-            if (idx + 2 >= length || end == idx)
+            if ((uint)(idx + 2) >= (uint)uriString.Length || end == idx)
```

Comment:
Is the JIT able to eliminate bounds checks like `uriString[idx + 1]` based on this uint casting?  I thought that was beyond its ability today.

Reply:
Ah, you are correct, it will eliminate it for `idx + 2` but not for `idx` or `idx + 1`.
I will try to eliminate these as well!